### DISCUSSION TITLE
Option to strip EXIF metadata before upload

### DISCUSSION THREAD

**Discussion by vishnukvmd** [ente-io/ente#896] [Enhancements]
**Describe the feature**
Settings > Advanced > Strip EXIF

**Comment by sans-c** [comment]
Related, albeit not the same: optionally strip EXIF metadata before sharing.

When I share photos, I potentally don't want to share all the metadata along with it. I realize this might be difficult to achieve in ente since you serve the original files. For this to work, the naive solution would be to create sanitized copies of shared photos first. Not ideal. Still something I'd very much like to see.

**Comment by whoami730** [comment]
With the new [embedding albums](https://help.ente.io/photos/features/sharing-and-collaboration/embed) support, I think this becomes even more necessary. I wish to upload/add my photos to the albums but I don't want detailed EXIF data such as device name, model, location etc to be shared.

Ideally, as @sans-c has mentioned above, it would be useful for there to be an option to strip EXIF metadata or not. Maybe public albums could default to not sharing EXIF data at all?

Without this though, I would need to unfortunately clone the image(s) to be able to embed them.

**Comment by kenny-qin** [comment]
I came here because I'm currently looking at trying the embed album feature that was recently released. Definitely would like to have safeguards in place around EXIF and sharing albums.

I wonder if it would be a good idea to be able to mark certain albums as "no metadata" and whenever you add something to the album, the EXIF will be stripped automatically. There should be a warning that extra storage will need to be used (unless there's a way around that).

**Comment by ailaG** [comment]
Extension to this suggestion: Option to strip some of the EXIF data only on public albums.
I may want people to know the date a photo was taken at but not the exact second or the exact GPS spot.

If we could control it in a high level too (e.g. code-like commands for which lines to remove, replace or round up the value of) even better.

**Comment by MayeulC** [comment]
I second this request. I also think that this option belongs on the public link settings page.

My use-case is stripping location information before sharing pictures. I don't really care about the other metadata.

A potential implementation would be using a different encryption key for the metadata section. If the key is not available when loading the image, strip metadata during decryption.
I think I will create a manual copy of the images for now and strip the metadata, but that's a big hassle, and hard to manage in the interface.
